src/toil/lib/docker.py

```diff
@@ -135,7 +135,7 @@ def toil_job(job):
                         stderr and/or stdout are True. Defaults to "output.log".
     :param dict log_config: Specify the logs to return from the container.  See:
                       https://docker-py.readthedocs.io/en/stable/containers.html
-    :param bool remove: Remove the container on exit or not.
+    :param bool remove: Remove the container on exit or not. (default: True).
```

Comment:
Ah, thanks for catching the problem with conflicting options. That was something that I didn't think about. 

And yes, I chose to use the `deferParam` system because `remove`/`auto_remove` doesn't work well with `detach=True`. So for example: 

```python
def docker_call(job):
    container = apiDockerCall(job,
                              image="ubuntu:latest",
                              parameters=["ls", "-lha", "/"],
                              detach=True,
                              remove=True,
                              )

    stdout = container.logs(stderr=False, stdout=True)  # 404 error!
    logger.warning(stdout)
```

Here, you'd get a `docker.errors.NotFound: 404 Client Error: Not Found ("No such container: "...")` error, because I think the container is removed before the `logs()` call. I'd be curious to know if there's a way to solve this? 

Using the `deferParams=RM` system, however, would call `docker kill` and `docker rm` after users are done with the container, so I think that would give the desired behavior? I originally planned to use the `remove` flag for clean up when `detach=False`, and defer remove when `detach=True`, but that seemed to complicate things more. 

I'm leaning towards fixing the default `deferParams` behavior, but let me know what you think.

Reply:
Hm, that is a good reason not to lean on `remove`/`auto_remove` for all our cleanup.

Fixing the default `deferParams` sounds like the right approach to me.